src/components/DetailContainer/DetailContainerTabs.tsx

```diff
@@ -0,0 +1,35 @@
+import React, { ChangeEvent, useState, useEffect } from 'react';
+import { default as MuiTabs } from '@material-ui/core/Tabs';
+import Tab from '@material-ui/core/Tab';
+import styled from 'react-emotion';
+
+import { TabPosition } from './tabs';
+
+interface Props {
+  tabPosition: TabPosition;
+  onChangeTabPosition: (event: ChangeEvent<{}>) => void;
+}
+
+const Tabs = styled(MuiTabs)({
+  marginBottom: 16,
+});
+
+const DetailContainerTabs: React.FC<Props> = ({ tabPosition, onChangeTabPosition }) => {
+  const [tabPositionIndex, setTabPositionIndex] = useState(0);
+
+  useEffect(() => {
+    const tabIndex = Object.keys(TabPosition).findIndex(position => position === String(tabPosition).toUpperCase());
+    setTabPositionIndex(tabIndex);
+  }, [tabPosition]);
```

Comment:
I think also `setTabPositionIndex` should be part of the list.

https://reactjs.org/docs/hooks-reference.html#useeffect

>  every value referenced inside the effect function should also appear in the dependencies array. In the future, a sufficiently advanced compiler could create this array automatically

Reply:
You can have `Object.keys(TabPosition)` outside of the component, it will never change anyway, this suggestion  is mostly an optimization.

Reply:
Hmmm are you sure? because the `setTabPositionIndex` won't change.

Reply:
I checked twice now and couldn't find any examples, also react official, having the setFunction in the array

Reply:
https://stackoverflow.com/a/55743744 👍